compiler/src/iree/compiler/Codegen/Common/VectorLayoutAnalysis.cpp

```diff
@@ -553,9 +554,27 @@ static void propagateLayoutToContractionOp(
     return;
   }
 
-  // True to resolve result with init.
-  ChangeResult changed = result->resolve(init);
-  update(result, changed);
+  if (!init->isUninitialized()) {
+    // True to resolve result with init.
+    ChangeResult changed = result->resolve(init);
+    update(result, changed);
+    return;
+  }
```

Comment:
What do we need to guard this under isUninitialized?

Reply:
What was i thinking... ? 
let me revert this

Reply:
ok it fails without that... 
coming back to your original question... why do we want to propogate init when if it does not have a layout?

Reply:
(i.e. it could be enforced from result . no ? )

Reply:
resolve->resolve will just return if there is no intialized layout. I was just thinking that you could do:

```
    ChangeResult changed = result->resolve(init);
    update(result, changed);
    if (changed) {
      return;
    }
 ```

its okay though, i was just curious why it was guarded. Feel free to choose what you prefer here.

Reply:
ah cool. I didnt realized need to check "changed". 
now changed.